blender/LilySurfaceScraper/Scrapers/PolyHavenHdriScraper.py

```diff
@@ -22,47 +22,65 @@
 # from a single URL
 
 from .AbstractScraper import AbstractScraper
-from urllib.parse import urlparse
+import requests
+import re
+import os
 
 class PolyHavenHdriScraper(AbstractScraper):
     scraped_type = {'WORLD'}
     source_name = "Poly Haven HDRI"
     home_url = "https://polyhaven.com/hdris"
+    home_dir = "hdrihaven"
+
+    polyHavenUrl = re.compile(r"(?:https:\/\/)?polyhaven\.com\/a\/([^\/]+)")
+
+    @classmethod
+    def getUid(cls, url):
+        match = cls.polyHavenUrl.match(url)
+        if match is not None:
+            return match.group(1)
+        return None
 
     @classmethod
     def canHandleUrl(cls, url):
         """Return true if the URL can be scraped by this scraper."""
-        return url.startswith("https://polyhaven.com/a")
```

Comment:
I'd rather keep `canHandleUrl` very fast (not requesting remote resource) and put this check in `fetchVariantList`. There is no risk that this scraper "steals" the URL from another one, if the URL points to an hdri it is meant to be rejected anyways, no other texture scraper will be interested in it.

Reply:
I changed it to this check because it was making an empty material when you put a texture link in the hdri one and vice versa

Reply:
Oh ok I understand. Then let's have the operator canceled when `fetchVariantList` returns an empty list (or raises an exception) but really no http request in here or it will make the add-on lag more than it already does on low quality internet connection!